### PR TITLE
Engine - update windows powershell integration

### PR DESCRIPTION
**Pull request by NahuFigueroa97**
|Related issue|
|---|
|#19407|

## Description

The existing Powershell integration in Wazuh needs to be updated and enhanced. With the evolution of Wazuh and the introduction of new syntax and standards, it's crucial to ensure that the Powershell integration aligns with these changes and extracts relevant fields efficiently.

### CODE REVIEW

**Review by soynof — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/integrations/windows/decoders/windows-powershell-operational.yml [hunk 1]
```diff
@@ -4,19 +4,18 @@ parents:
...
   author:
     name: Wazuh, Inc.
-    date: 2023/01/23
+    date: 2023/10/10
   references:
     - https://learn.microsoft.com/en-us/powershell/scripting/windows-powershell/wmf/whats-new/script-logging?view=powershell-7.3
-  versions: [Vista, "7", "8", "10", "11", Server 2012, Server 2016, Server 2019, Server 2022]
 
-check: string_equal($windows.System.Channel.#text,Microsoft-Windows-PowerShell/Operational)
+check: $windows.System.Channel.#text == Microsoft-Windows-PowerShell/Operational
```
Comment (soynof) [inline src/engine/ruleset/integrations/windows/decoders/windows-powershell-operational.yml:18]: We should unify the way of checking this field. I think the most legible way is how it was added in the `Windows Powershell` decoder, like this:

```
check:
  - windows.System.Channel.#text: Microsoft-Windows-PowerShell/Operational
```

What do you think?

**Review by soynof — CHANGES_REQUESTED** [review]

**Comment by NahuFigueroa97** [comment]
The values of system keywords and opcode were obtained from here https://github.com/nihon-tc/Rtest/blob/8246e4d21323802fb84c406edc1a005991304f5a/header/Microsoft%20SDKs/Windows/v7.0A/Include/winmeta.xml
Also, because the keywords values are 64 bits, the helper was updated with said change.

**Review by JavierBejMen — CHANGES_REQUESTED** [review]
The function was 32-bit because where used on 32-bit masks, please revert changes and make an additional function or template.

File: src/engine/source/builder/builders/opBuilderKVDB.cpp [hunk 2]
```diff
@@ -578,11 +578,11 @@ HelperBuilder getOpBuilderKVDBGetArray(std::shared_ptr<IKVDBManager> kvdbManager
 namespace
 {
 // TODO Change this to use an vector instead of a map
-std::function<std::optional<json::Json>(uint32_t pos)> getFnSearchMap(const json::Json& jMap)
+std::function<std::optional<json::Json>(uint64_t pos)> getFnSearchMap(const json::Json& jMap)
```
Comment (soynof) [inline src/engine/source/builder/builders/opBuilderKVDB.cpp:581]: Shouldn't we have both functions? one for 32 and one for 64 bits?

File: src/engine/test/source/builder/builders/opBuilderKVDBDecodeBitmask_test.cpp [hunk 3]
```diff
@@ -152,8 +152,8 @@ INSTANTIATE_TEST_SUITE_P(
         BuildMapT(json::Json {R"( {"0" : "zero", "1" : "one", "2" : "two", "3" : ["null"]} )"}, false),
         // Map with keys out of range
         BuildMapT(json::Json {R"( {"-1" : "zero", "1" : "one"} )"}, false),
-        BuildMapT(json::Json {R"( {"0" : "zero", "1" : "one", "32" : "thirty two"} )"}, false),
-        BuildMapT(json::Json {R"( {"0" : "zero", "1" : "one", "32" : "thirty two", "33" : "thirty three"} )"}, false),
+        BuildMapT(json::Json {R"( {"0" : "zero", "1" : "one", "64" : "sixty four"} )"}, false),
+        BuildMapT(json::Json {R"( {"0" : "zero", "1" : "one", "64" : "sixty four", "65" : "sixty five"} )"}, false),
```
Comment (soynof) [inline src/engine/test/source/builder/builders/opBuilderKVDBDecodeBitmask_test.cpp:156]: We should have coverage for 32 and 64 bit masks

File: src/engine/test/source/builder/builders/opBuilderKVDBDecodeBitmask_test.cpp [hunk 4]
```diff
@@ -269,11 +269,11 @@ INSTANTIATE_TEST_SUITE_P(
...
-        DecodeMaskT(R"("0x100000000")", {}, false),
-        DecodeMaskT(R"("0x100000001")", {}, false),
-        DecodeMaskT(R"("0x100000002")", {}, false),
-        DecodeMaskT(R"("0x100000003")", {}, false),
-        DecodeMaskT(R"("0x1FFFFFFFF")", {}, false),
+        DecodeMaskT(R"("0x100000000BB672397")", {}, false),
+        DecodeMaskT(R"("0x100000001492637AA")", {}, false),
+        DecodeMaskT(R"("0x1000000026B881A11")", {}, false),
+        DecodeMaskT(R"("0x1000000031A021C11")", {}, false),
+        DecodeMaskT(R"("0x1FFFFFFFFFFFFFFFF")", {}, false),
```
Comment (soynof) [inline src/engine/test/source/builder/builders/opBuilderKVDBDecodeBitmask_test.cpp:276]: ditto

**Review by soynof — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/integrations/windows/decoders/windows-powershell.yml [hunk 5]
```diff
@@ -71,14 +70,26 @@ normalize:
...
 
-# Cleanup
+# windows.logon
  - map:
-   - windows: delete()
+   - windows.logon.activity_id: $windows.System.Task.@ActivityID
+   - windows.logon.channel: $windows.System.Channel.#text
+   - windows.logon.computer_name: $windows.System.Computer.#text
+   - windows.logon.event_id: $windows.System.EventID.#text
+   - windows.logon.keywords: kvdb_decode_bitmask(windows_bitmask_tables, Keywords, $windows.System.Keywords.#text)
```
Comment (juliancnn) [inline src/engine/ruleset/integrations/windows/decoders/windows-powershell.yml:87]: Should  check the special case wjhens Keywords, `$windows.System.Keywords.#text == 0x0`

**Review by juliancnn — CHANGES_REQUESTED** [review]

**Review by soynof — APPROVED** [review]
LGTM!

**Review by juliancnn — APPROVED** [review]

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**